### PR TITLE
Added mock of core-storage api;

### PR DESCRIPTION
**Pull request by koerberm**
Re-activated tests by mocking core-storage api;

### CODE REVIEW

**Comment by koerberm** [comment]
[NFDI-Core-Storage-Provider: API mocken in Tests](https://trello.com/c/45NWBKOR/518-nfdi-core-storage-provider-api-mocken-in-tests)

File: services/src/datasets/external/nfdi/mod.rs [hunk 1]
```diff
@@ -665,23 +665,56 @@ mod tests {
...
+    use tokio::io::AsyncReadExt;
+
+    use crate::datasets::listing::{DatasetListOptions, ExternalDatasetProvider, OrderBy};
+    use crate::util::user_input::Validated;
+
+    mod wiremock_gen {
+        // hello.Greeter: is the prefix of all rpc,
+        // MyMockServer: name of the generated Server,
+        wiremock_grpc::generate!("api.services.v1", TestProjectServer);
+    }
```
Comment (ChristianBeilschmidt) [inline services/src/datasets/external/nfdi/mod.rs:691]: It looks strange that the `mod` definition is surrounded by all the imports. I guess it should be after the imports.

File: services/Cargo.toml [hunk 2]
```diff
@@ -75,6 +75,7 @@ rand = "0.8.4"
 tempfile = "3.1"
 walkdir = "2.3"
 xml-rs = "0.8.3"
+wiremock-grpc = "0.0.3-alpha1"
```
Comment (ChristianBeilschmidt) [inline services/Cargo.toml:78]: What is the advantage of using `wiremock-grpc` over our current mock web server?

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: services/src/datasets/external/nfdi/mod.rs (hunk 1 above)
Comment (koerberm) [reply services/src/datasets/external/nfdi/mod.rs]: 96a84faeea669c364687b09127ee5ba955cc163f
But the imports of the generated code must occur after the module definition;

File: services/Cargo.toml (hunk 2 above)
Comment (koerberm) [reply services/Cargo.toml:77]: The regular httptest does not implement the gRPC protocol and serialization.

**Review by ChristianBeilschmidt — APPROVED** [review]

**Comment by ChristianBeilschmidt** [comment]
bors merge

**Pull request merged**